openmc/universe.py

```diff
@@ -701,13 +705,24 @@ def get_unique_universes(self):
 
         """
 
-        unique_universes = np.unique(self._universes.ravel())
-        universes = {}
-
-        for universe in unique_universes:
-            universes[universe._id] = universe
-
-        return universes
+        univs = dict()
```

Comment:
I don't want to make too big of a deal here, but this method can be *very* useful for tally data post-processing in parametric studies, and its performance can be a major bottleneck. As I discovered the hard way with some profiling of my OpenCG/OpenMC tally data processing workflow earlier this year, using ``numpy.unique(...)`` is *much, much* faster than iterating through a list of lists like this (as one would expect). This is one reason I hesitate to support the idea of storing ``Universes`` in lists. Anyway, if we can think of a way to simplify and speed this routine up using NumPy, smart caching, or something else altogether that would be nice.